### PR TITLE
Update overdrive formats when a delivery mechanism is no longer available

### PR DESCRIPTION
**Pull request by aslagle**
When we are unable to lock in a format that we thought was available, we will now update the license pool delivery mechanisms and give the user a specific 502. 

There's also a sweep script to update formats for all the overdrive books in our collection.

### CODE REVIEW

File: api/overdrive.py [hunk 1]
```diff
@@ -262,7 +278,12 @@ def get_fulfillment_link(self, patron, pin, overdrive_id, format_type):
             response = self.lock_in_format(
                 patron, pin, overdrive_id, format_type)
             if response.status_code not in (201, 200):
-                raise CannotFulfill("Could not lock in format %s" % format_type)
+                if response.status_code == 400:
+                    message = json.loads(response.content).get("message")
+                    if message == "The selected format may not be available for this title.":
+                        raise FormatNotAvailable("This book is not Available the requested format")
```
Comment (leonardr) [inline api/overdrive.py]: Copyedit: "This book is not available in the format you requested."

I don't normally copyedit but this string will go out to patrons.

File: tests/files/overdrive/lock_in_format_not_available.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"errorCode":"PatronTitleProcessingFailed","message":"The selected format may not be available for this title.","token":"bf3b1876-20fa-4755-a923-acc809740002"}
```
Comment (leonardr) [inline tests/files/overdrive/lock_in_format_not_available.json]: This is what Overdrive says? That's so wishy-washy.

**Comment by leonardr** [comment]
:+1: after a copyediting change.

File: api/overdrive.py (hunk 1 above)
Comment (aslagle) [inline api/overdrive.py]: :+1:

File: tests/files/overdrive/lock_in_format_not_available.json (hunk 2 above)
Comment (aslagle) [inline tests/files/overdrive/lock_in_format_not_available.json]: It is :(

**Pull request merged**